### PR TITLE
Link customer display to register with pairedDeviceId

### PR DESCRIPTION
**Pull request by jchirico8**
### Issues Fixed
[PCOM-3058](https://jira.ae.com/browse/PCOM-3058) - Update DeviceModel to Support Pairing Customer Display
[PCOM-3061](https://jira.ae.com/browse/PCOM-3061) - Add pairedDeviceId to AppEvent
[PCOM-3063](https://jira.ae.com/browse/PCOM-3063) - Update @OnEvent to Receive Events from Paired Device
[PCOM-3066](https://jira.ae.com/browse/PCOM-3066) - Add a CustomerDisplayManager that interacts with the StateManagerContainer
[PCOM-3069](https://jira.ae.com/browse/PCOM-3069) - Update Status Bar Dialog for Pairing Customer Display

### Summary
Adding support for a customer display that does not have the same id as the base register. These 2 devices will be linked by the pairedDeviceId property, which I have attached to AppEvent. There is also a new property to OnEvent called `receiveEventsFromPairedDevice` so we control when to handle events from the paired device.

### Database Changes
Adds a pairedDeviceId column to DeviceModel

### CODE REVIEW

**Review by chenson42 — APPROVED** [review]

File: openpos-util/src/main/java/org/jumpmind/pos/util/event/OnEvent.java [hunk 1]
```diff
@@ -36,6 +36,11 @@
      */
     boolean receiveEventsFromSelf() default false;
 
+    /**
+     * TODO describe
```
Comment (kdelorey) [inline openpos-util/src/main/java/org/jumpmind/pos/util/event/OnEvent.java:40]: Any reason why we can't describe this now?

File: openpos-devices/src/main/java/org/jumpmind/pos/devices/model/DevicesRepository.java [hunk 2]
```diff
@@ -36,7 +39,7 @@ public DeviceModel getDevice(String deviceId, String appId) {
             if (virtualDevice != null) {
                 return virtualDevice;
             }
-            throw new DeviceNotFoundException("No device found for appId=" + appId + " deviceId=" + deviceId);
+            return null;
```
Comment (kdelorey) [inline openpos-devices/src/main/java/org/jumpmind/pos/devices/model/DevicesRepository.java:42]: Are there any usages of `getDevice(String, String)` that who's logic is now affected by this change? I didn't notice any related changes in this PR.

**Review by kdelorey — APPROVED** [review]
There's something worth verifying, other than that, lgtm.

File: openpos-util/src/main/java/org/jumpmind/pos/util/event/OnEvent.java (hunk 1 above)
Comment (jchirico8) [reply openpos-util/src/main/java/org/jumpmind/pos/util/event/OnEvent.java:40]: The best reason, I forgot. I'll get something in there

**Review by kdelorey — APPROVED** [review]

**Review by chenson42 — APPROVED** [review]

**Pull request merged**